sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -479,6 +484,21 @@ class depends_on {
 }
 ----
 
+==== Depends-On-All-Leaves Property 
+The API for explicitly adding nodes to a `command_graph` includes a
+`property_list` parameter. This extension defines the `depends_on_all_leaves` 
+property to be passed here. `depends_on_all_leaves` provides a shortcut for 
+adding all leaves of a graph as dependencies.
+Therefore the created node will depend on all the current leaves in the graph.  
```

Comment:
```suggestion
adding all leaves of a graph as dependencies of the newly created node.
```

Reply:
Do we need to specify when we actually evaluate the node dependencies? For a multi-threaded scenario the set of leaves wouldn't necessarily be constant.

Reply:
I merged the two sentences in one but I kept the notion of "current" leaves.

Reply:
IMO, in a multi-threading scenario this property could lead to an indeterminate graph since the order of adding nodes is not deterministic, but it cannot lead to an invalid state of the graph.